tests/ui/mismatched_types/hr-projection-mismatch.next.stderr

```diff
@@ -0,0 +1,20 @@
+error[E0271]: type mismatch resolving `<Thing as Trait<'a>>::Assoc == &i32`
```

Comment:
weird error message lol

Reply:
but unrelated, probably would be nice to have a leak-check specific error message or sth.